terraso_backend/apps/soil_id/models/depth_dependent_soil_data.py

```diff
@@ -13,13 +13,273 @@
+    (PH_TESTING_INDICATOR_SOLUTION, _("pH indicator solution")),
+    (PH_TESTING_METER, _("pH meter")),
+    (OTHER, _("other")),
+)
+
+SOIL_TESTING_DRY_COMBUSTION = "Dry combustion"
+SOIL_TESTING_WET_OXIDATION = "Wet oxidation (Walkey-Black)"
+SOIL_TESTING_LOSS_ON_IGNITION = "Loss-on-ignition"
+SOIL_TESTING_REFLECTANCE_SPECTROSCOPY = "Reflectance spectroscopy"
+SOIL_TESTING_FIELD_REFLECTOMETER = "Field reflectometer"
+
+SOIL_TESTING_METHODS = (
+    (SOIL_TESTING_DRY_COMBUSTION, _("Dry combustion")),
+    (SOIL_TESTING_WET_OXIDATION, _("Wet oxidation (Walkey-Black)")),
+    (SOIL_TESTING_LOSS_ON_IGNITION, _("Loss-on-ignition")),
+    (SOIL_TESTING_REFLECTANCE_SPECTROSCOPY, _("Reflectance spectroscopy")),
+    (SOIL_TESTING_FIELD_REFLECTOMETER, _("Field reflectometer")),
+    (OTHER, _("Other")),
+)
+
 
 class DepthDependentSoilData(BaseModel):
     soil_data_input = models.ForeignKey(SoilData, on_delete=models.CASCADE)
     depth_top = models.IntegerField(blank=True)
     depth_bottom = models.IntegerField(blank=True)
+    texture = models.CharField(null=True, choices=TEXTURES)
+    rock_fragment_volume = models.CharField(null=True, choices=ROCK_FRAGMENT_VOLUMES)
+    color_hue = models.CharField(null=True, choices=COLOR_HUES)
+    color_hue_substep = models.CharField(null=True, choices=COLOR_HUES_SUBSTEPS)
+    color_value = models.CharField(null=True, choices=COLOR_VALUES)
+    color_chroma = models.CharField(null=True, choices=COLOR_CHROMAS)
+    conductivity = models.DecimalField(
+        null=True, max_digits=100, decimal_places=2, validators=[MinValueValidator(0)]
+    )
+    conductivity_test = models.CharField(null=True, choices=CONDUCTIVITY_TESTS)
+    conductivity_unit = models.CharField(null=True, choices=CONDUCTIVITY_UNITS)
+    structure = models.CharField(null=True, choices=STRUCTURES)
+    ph = models.DecimalField(
+        null=True,
+        max_digits=3,
+        decimal_places=1,
+        validators=[MinValueValidator(0.0), MaxValueValidator(14.0)],
+    )
+    ph_testing_solution = models.CharField(null=True, choices=PH_TESTING_SOLUTIONS)
+    ph_testing_method = models.CharField(null=True, choices=PH_TESTING_METHODS)
+    soil_organic_carbon = models.DecimalField(
+        null=True,
+        max_digits=100,
+        decimal_places=1,
+        validators=[MinValueValidator(0), MaxValueValidator(100)],
+    )
+    soil_organic_matter = models.DecimalField(
+        null=True,
+        max_digits=100,
+        decimal_places=1,
+        validators=[MinValueValidator(0), MaxValueValidator(100)],
+    )
+    soil_organic_carbon_testing = models.CharField(null=True, choices=SOIL_TESTING_METHODS)
+    soil_organic_matter_testing = models.CharField(null=True, choices=SOIL_TESTING_METHODS)
+    sodium_absorption_ratio = models.DecimalField(
+        null=True, max_digits=100, decimal_places=2, validators=[MinValueValidator(0)]
+    )
+    bedrock = models.PositiveIntegerField(null=True)
```

Comment:
nit: i think 0 is a valid value for bedrock, so `IntegerField` with `MinValueValidator(0)` might make more sense

Reply:
@shrouxm I think `PositiveIntegerField` includes 0: https://www.geeksforgeeks.org/positiveintegerfield-django-models/